File: shared/android/app/src/main/java/io/keybase/ossifrage/modules/BackgroundSyncWorker.java
package io.keybase.ossifrage.modules;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import keybase.Keybase;

public class BackgroundSyncWorker extends Worker {
    public static final String TAG = "background_sync_job";

    public BackgroundSyncWorker(
            @NonNull Context context,
            @NonNull WorkerParameters params) {
        super(context, params);
    }

    @Override
    public Result doWork() {
        Log.d(TAG, "Background sync start.");
        Keybase.backgroundSync();
        Log.d(TAG, "Background sync complete.");
        return Result.success();
    }
}

File: shared/android/app/src/main/java/io/keybase/ossifrage/modules/BackgroundSyncWorker.kt
package io.keybase.ossifrage.modules

import android.content.Context
import android.util.Log
import androidx.work.Worker
import androidx.work.WorkerParameters
import keybase.Keybase

class BackgroundSyncWorker(
        context: Context,
        params: WorkerParameters) : Worker(context, params) {
    override fun doWork(): Result {
        Log.d(TAG, "Background sync start.")
        val result = Keybase.backgroundSync()
        Log.d(TAG, "Background sync complete: $result")
        return Result.success()
    }

    companion object {
        const val TAG = "background_sync_job"
    }
}
